### ISSUE TITLE
loadTable() should support a separator specification option

### ISSUE THREAD

**Issue by tbarnier** [processing/processing#5068]
it is impossible to import a csv file where fields are separated by a non standard character (like ';', or '|').

a 'sep' option allowing to specify a field delimiter character would help usability, and be in line with other framework like in R, or in python pandas read_csv()

**Comment by jeremydouglass** [comment]
To support non-standard delimiters (that don't also appear in field data), preprocess the file with search-replace. For previous discussion and this workaround, see: #3039

**Comment by benfry** [comment]
I'll mark this as an enhancement. My hope was to keep things simple by not supporting every CSV variant, but this is coming up frequently enough that the current limitation seems to undermine the usefulness of the `Table` class. 

Also note that OpenOffice `.ods` files also work with `loadTable()`, which provide a more robust format.

Help requested for the delimiter implementation…

**Comment by TheGuywithTheHat** [comment]
I've looked over the code, and I'll try to get this done within a few days. Should I add the customization to saveTable() as well?

**Comment by benfry** [comment]
Yeah… The tricky thing here is how alternate delimiters handle escapes. So if you're using | or ; we need to make sure the escaped versions of the same work properly (and make sure we're aware of the correct escaping systems).

**Comment by christianbender** [comment]
How about a simple class for desired functionality?

**Comment by christianbender** [comment]
I build a CSV class (in Processing itself) to support the requests.

**Comment by TheGuywithTheHat** [comment]
@christianbender I got started on this, but ended up too busy to finish. You can see what changes I made at https://github.com/TheGuywithTheHat/processing/commits/load-table. Feel free to pull from my repo (load-table branch) and build off of what I have, or just use my code as a guideline.

I think what I was stuck on last summer was how to specify the delimiter:

 - if it's part of the options string, we would have to either disallow certain delimiters or implement our own escape sequence parser.
 - if it's a parameter of its own, we would have to somehow handle cases where the user doesn't want to input an options string.

**Comment by benfry** [comment]
Thanks for taking a look at it (and sounds like you're starting to understand the trickiness I was referring to…)

**Comment by christianbender** [comment]
@TheGuywithTheHat  Thanks for response. I see it at the moment.
